pkg/cli/mirror/catalog_images.go

```diff
@@ -233,6 +310,249 @@ func (o *MirrorOptions) processCatalogRefs(ctx context.Context, catalogsByImage
 	return nil
 }
 
+// findOpmCmd attempts to find the opm binary within the extracted contents of the catalog container image.
+// The exact location of the opm binary within the image is not certain: It depends among other things on
+// the version of the catalog, on the decisions by maintainers of the catalog, on the platform...
+func findOpmCmd(artifactDir string) (string, error) {
+	wd, err := os.Getwd()
+	if err != nil {
+		return "", fmt.Errorf("error finding current working directory while preparing to run opm to regenerate cache: %v", err)
+	}
+	runningOS := runtime.GOOS
+	runningArch := runtime.GOARCH
+	opmBin := "opm"
+
+	if runningOS != "linux" {
+		opmBin = strings.Join([]string{runningOS, runningArch, opmBin}, "-")
+	}
+	opmCmdPath := ""
+	binaryDir := filepath.Join(wd, artifactDir, config.OpmBinDir)
+	err = filepath.Walk(binaryDir, func(path string, info fs.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+
+		// files that match the opmBin name with size 0 are the result of extracted symbolic links where the link has broken during extraction
+		if info.Name() == opmBin && info.Size() > 0 {
+			opmCmdPath = path
+		}
+
+		return nil
+	})
+	if err != nil {
+		return "", fmt.Errorf("error finding the extracted opm binary %s while preparing to run opm to regenerate cache: %v", opmCmdPath, err)
+	}
+	_, err = os.Stat(opmCmdPath)
+	if err != nil {
+		return "", fmt.Errorf("error finding the extracted opm binary %s while preparing to run opm to regenerate cache: %v", opmCmdPath, err)
+	}
+	err = os.Chmod(opmCmdPath, 0744)
+	if err != nil {
+		return "", fmt.Errorf("error changing permissions to the extracted opm binary while preparing to run opm to regenerate cache: %v", err)
+	}
+	return opmCmdPath, nil
+}
+
+// extractOPMAndCache is usually called after rendering catalog's declarative config.
+// it uses crane modules to pull the catalog image, select the manifest that corresponds to the
+// current platform. It then extracts from that image any files that are suffixed `*opm` for later
+// use upon rebuilding the catalog: This is because the opm binary can be called `opm` but also
+// `darwin-amd64-opm` etc.
+func extractOPMAndCache(ctx context.Context, srcRef image.TypedImageReference, outDir string, insecure bool) error {
```

Comment:
Nit: the  `outDir` seems to be the catalog source directory... maybe it should be called that for clarity?